### PR TITLE
fix: agreement creation flow

### PR DESCRIPTION
**Pull request by offerijns**

### CODE REVIEW

File: onboard-api/src/controllers/address.controller.ts [hunk 1]
```diff
@@ -66,27 +67,39 @@ export class AddressController {
...
-      )
-
-      const agreementLinks = agreements.map(
-        (agreement: Agreement): AgreementsStatus => {
+      // Filter profile agreements by country
+      const profileAgreements = pool.profile?.agreements
+        .filter((pa: ProfileAgreement) => {
+          return (
+            (user.countryCode === 'US' && pa.country === 'non-us') || (user.countryCode !== 'US' && pa.country === 'us')
+          )
```
Comment (philipstanislaus) [inline onboard-api/src/controllers/address.controller.ts:75]: Shall we throw an error here if these filters don't match?

File: onboard-api/src/controllers/address.controller.ts [hunk 2]
```diff
@@ -66,27 +67,39 @@ export class AddressController {
...
+      // Retrieve
+      const agreementLinks = await this.agreementRepo.getStatusForProfileAgreements(
+        user.id,
+        params.poolId,
+        profileAgreements
       )
 
       const isWhitelisted = await this.investmentRepo.getWhitelistStatus(address.id, params.poolId)
 
+      // TODO: this should also filter by blockchain and network
```
Comment (philipstanislaus) [inline onboard-api/src/controllers/address.controller.ts:95]: Should we create a ticket for this? Could be a "good first issue" for one of our new hires.

File: onboard-api/src/services/pool.service.ts [hunk 3]
```diff
@@ -51,7 +54,14 @@ export class PoolService {
     )
 
     this.pools = poolsWithProfiles
-    this.logger.log(`Loaded ${Object.keys(this.pools).length} pools with profiles from IPFS`)
+    const newPools = Object.values(poolsWithProfiles).filter((pool: Pool) => !prevPools.includes(pool))
+
+    if (newPools.length > 0) this.logger.log(`Loaded ${Object.keys(this.pools).length} pools with profiles from IPFS`)
```
Comment (philipstanislaus) [inline onboard-api/src/services/pool.service.ts:59]:
```suggestion
    if (newPools.length > 0) this.logger.log(`Loaded ${Object.keys(this.pools).length} new pools with profiles from IPFS`)
```

File: onboard-api/src/controllers/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { Agreement, AgreementRepo } from '../repos/agreement.repo'
+import { UserRepo, UserWithKyc } from '../repos/user.repo'
+
+@Controller()
+export class UserController {
+  constructor(private readonly agreementRepo: AgreementRepo, private readonly userRepo: UserRepo) {}
+
+  @Get('users/:poolId')
+  async getUsers(@Param() params) {
+    return {}
```
Comment (philipstanislaus) [inline onboard-api/src/controllers/user.controller.ts:12]: Is a TODO missing here?

**Review by philipstanislaus — APPROVED** [review]
LGTM, just a few minor comments 👍

File: tinlake.js/bin/decode.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+const run = async () => {
+  const decoded = contract.interface.decodeFunctionData('borrow', '0x0ecbcdab0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000675e56da063e8940000')
+  decoded.forEach((value) => {
+    console.log(value.toString())
+  })
+
+}
+
+run()
```
[the remark this answers is not in the document]
Comment (offerijns) [reply tinlake.js/bin/decode.js:21]: Why not? I think it's a useful debugging script, that I imagine I'll definitely also use in the future.

[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake.js/bin/decode.js:21]: Okay :)

File: onboard-api/src/controllers/address.controller.ts (hunk 1 above)
Comment (offerijns) [reply onboard-api/src/controllers/address.controller.ts:75]: That wouldn't be an error, this is just a filter to get the US agreement for US based investors and vice versa.

File: onboard-api/src/controllers/address.controller.ts (hunk 2 above)
Comment (offerijns) [reply onboard-api/src/controllers/address.controller.ts:95]: I would say touching this codebase shouldn't be a first issue actually, it requires quite a lot of understanding of the onboarding process and how all the integrations fit together.

File: onboard-api/src/controllers/user.controller.ts (hunk 4 above)
Comment (offerijns) [reply onboard-api/src/controllers/user.controller.ts:12]: Yes :)

**Pull request merged**